src/d/actor/d_a_obj_eff.cpp

```diff
@@ -2,261 +2,479 @@
+    int ret;
+    cXyz pScale;
+    pScale.x = scale.x;
+    pScale.y = scale.y;
+    pScale.z = scale.z;
+
+    dPa_control_c *particle = g_dComIfG_gameInfo.play.getParticle();
+    JPABaseEmitter *emitter =
+            particle->set(dPa_control_c::dPtclGroup_Toon_e, dPa_name::ID_COMMON_2027, &current.pos, NULL, &pScale, 0xb4,
+                          mParticleCallback, -1, NULL, NULL, NULL);
+
+    if (emitter != NULL) {
+        emitter->mVolumeSize = 0x1E;
+        emitter->mLifeTime = 0x23;
+        emitter->mRate = 20.0f;
+        emitter->mMaxFrame = 1;
+        emitter->mInitialVelOmni = 10.0f;
+        static JGeometry::TVec3<f32> em_scl(1.0f, 0.0f, 1.0f);
+        emitter->setEmitterScale(em_scl);
+        static JGeometry::TVec3<f32> em_trans(0.0f, 5.0f, 0.0f);
+        emitter->setEmitterTranslation(em_trans);
+        ret = true;
+    } else {
+        ret = false;
+    }
+
+    return ret;
 }
 
 /* 00000D24-00000D98       .text eff_set_pinecone_smoke__Q28daObjEff5Act_cFv */
-void daObjEff::Act_c::eff_set_pinecone_smoke() {
-    /* Nonmatching */
+BOOL daObjEff::Act_c::eff_set_pinecone_smoke() {
+    dPa_control_c *particle = g_dComIfG_gameInfo.play.getParticle();
+    JPABaseEmitter *emitter =
+            particle->set(dPa_control_c::dPtclGroup_Toon_e, dPa_name::ID_SCENE_A16B, &current.pos, NULL, NULL, 0xb4,
+                          mParticleCallback, -1, NULL, NULL, NULL);
+
+    return emitter != NULL;
 }
 
 /* 00000D98-00000E80       .text eff_set_woodBox_smoke__Q28daObjEff5Act_cFv */
-void daObjEff::Act_c::eff_set_woodBox_smoke() {
-    /* Nonmatching */
+BOOL daObjEff::Act_c::eff_set_woodBox_smoke() {
+    static cXyz particle_scl(2.5f, 2.5f, 1.0f);
+    dPa_control_c *particle = g_dComIfG_gameInfo.play.getParticle();
+    JPABaseEmitter *emitter = particle->set(dPa_control_c::dPtclGroup_Toon_e, dPa_name::ID_COMMON_2027, &current.pos,
+                                            NULL, NULL, 0xb4, mParticleCallback, -1, NULL, NULL, &particle_scl);
+
+    if (emitter != NULL) {
+        emitter->mRate = 30.0f;
+        emitter->mMaxFrame = 1;
+        return true;
+    }
+
+    return false;
 }
 
 /* 00000E80-00000F6C       .text eff_set__Q28daObjEff5Act_cFv */
-void daObjEff::Act_c::eff_set() {
-    /* Nonmatching */
+BOOL daObjEff::Act_c::eff_set() {
+    static Proc proc[] = {
+        &eff_set_barrel_smoke,
+        &eff_set_stool_smoke,
+        &eff_set_skull_smoke,
+        &eff_set_land_smoke,
+        &eff_set_pinecone_smoke,
+        &eff_set_woodBox_smoke
+    };
+
+    return (this->*proc[mProcIndex])();
 }
 
 /* 00000F6C-00001030       .text _create__Q28daObjEff5Act_cFv */
 cPhs_State daObjEff::Act_c::_create() {
-    /* Nonmatching */
+    static const u32 heap_size[] = {
+        0x00000040,
+        0x00000040,
+        0x00000040,
+        0x00000020,
+        0x00000040,
+        0x00000040,
+    };
+
+    mProcIndex = daObj::PrmAbstract(this, PRM_ARG_W, PRM_ARG_S);
```

Comment:
Should use the inline from the debug map.
```suggestion
    mProcIndex = prm_get_type();
```
And `PRM_ARG` should be renamed to `PRM_TYPE` to match the official name.

Reply:
Updated enum names and changed to use prm_get_type method.